### PR TITLE
feat: extend flux-dsl

### PR DESCRIPTION
**Pull request by Andy2003**
## Proposed Changes

* Add ability to define imports for each flux function
* Add ability use multiple flux expressions
* Add ability to define custom functions
* Improve join flux, so it can be nested
* Add missing parameter variants for RangeFlux amd AggregateWindow
* Add TruncateTimeColumnFlux
* Add ArrayFromFlux
* Add UnionFlux

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [x] A test has been added if appropriate
- [x] `mvn test` completes successfully
- [x] Commit messages are [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

### CODE REVIEW

File: flux-dsl/src/main/java/com/influxdb/query/dsl/Flux.java [hunk 1]
```diff
@@ -1373,6 +1416,29 @@ public final RangeFlux range(@Nonnull final Long start, @Nonnull final Long stop
...
 
+    /**
+     * Filters the results by time boundaries.
+     *
+     * @param start Specifies the oldest time (Unix timestamp in seconds) to be included in the results
+     * @param stop  Specifies the exclusive newest time (Unix timestamp in seconds) to be included in the results
+     * @return {@link RangeFlux}
+     */
+    @Nonnull
+    public final RangeFlux range(final long start, final long stop) {
```
Comment (bednar) [inline flux-dsl/src/main/java/com/influxdb/query/dsl/Flux.java:1427]: Please use `Long` object:

```suggestion
    public final RangeFlux range(final Long start, final Long stop) {
```

File: flux-dsl/src/main/java/com/influxdb/query/dsl/Flux.java [hunk 2]
```diff
@@ -1373,6 +1416,29 @@ public final RangeFlux range(@Nonnull final Long start, @Nonnull final Long stop
...
+    }
+
+    /**
+     * Filters the results by time boundaries.
+     *
+     * @param start Specifies the oldest time (Unix timestamp in seconds) to be included in the results
+     * @return {@link RangeFlux}
+     */
+    @Nonnull
+    public final RangeFlux range(final long start) {
```
Comment (bednar) [inline flux-dsl/src/main/java/com/influxdb/query/dsl/Flux.java:1438]: Please use `Long` object:

```suggestion
    public final RangeFlux range(final Long start) {
```

File: flux-dsl/src/main/java/com/influxdb/query/dsl/HasImports.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.influxdb.query.dsl;
+
+import java.util.Set;
+
+public interface HasImports {
```
Comment (bednar) [inline flux-dsl/src/main/java/com/influxdb/query/dsl/HasImports.java:26]: Please add simple `javadoc` description.

File: flux-dsl/src/main/java/com/influxdb/query/dsl/functions/RangeFlux.java [hunk 4]
```diff
@@ -106,6 +106,18 @@ public RangeFlux withStart(@Nonnull final String start) {
         return this;
     }
 
+    /**
+     * @param start Specifies the oldest time (Unix timestamp in seconds) to be included in the results
+     * @return this
+     */
+    @Nonnull
+    public RangeFlux withStart(final long start) {
```
Comment (bednar) [inline flux-dsl/src/main/java/com/influxdb/query/dsl/functions/RangeFlux.java:114]: Please use `Long` object:

```suggestion
    public RangeFlux withStart(final Long start) {
```

File: flux-dsl/src/main/java/com/influxdb/query/dsl/functions/RangeFlux.java [hunk 5]
```diff
@@ -149,4 +161,17 @@ public RangeFlux withStop(@Nonnull final String stop) {
 
         return this;
     }
+
+    /**
+     * @param stop Specifies the exclusive newest time (Unix timestamp in seconds) to be included in the results
+     * @return this
+     */
+    @Nonnull
+    public RangeFlux withStop(final long stop) {
```
Comment (bednar) [inline flux-dsl/src/main/java/com/influxdb/query/dsl/functions/RangeFlux.java:170]: Please use `Long` object:

```suggestion
    public RangeFlux withStop(final Long stop) {
```

**Review by bednar — CHANGES_REQUESTED** [review]
Hi @Andy2003, thanks for your PR 👍 

Can you satisfy following requirements: 
1. Add new functions into `flux-dsl/README.md`:
    - `InterpolateLinearFlux`
    - `TruncateTimeColumnFlux`
2. Add description how to use the `imports` into `flux-dsl/README.md`
3. Add description how to use the `AbstractFunctionFlux` into `flux-dsl/README.md`

and also:

**Review by bednar — APPROVED** [review]
LGMT 🚀 

Thanks again for your PR 👍. From my POV is ready to merge into `master`. Do you want to continue in work or it is done?

**Comment by Andy2003** [comment]
> Do you want to continue in work or it is done?

No it's all done, please merge!

**Pull request merged**